azure-pipelines.yml

```diff
@@ -1,342 +1,424 @@
-                  -target azurerm_role_assignment.preview_externaldns_read_rg["\"${{parameters.cluster}}\""]
-                  -target azurerm_role_assignment.preview_externaldns_dns_zone_contributor["\"${{parameters.cluster}}\""]
-                  -target azurerm_role_assignment.preview2aat_cft_rg_identity_operator["\"${{parameters.cluster}}\""]
-                  -target data.azurerm_kubernetes_cluster.kubernetes["\"${{parameters.cluster}}\""]
-
-  - stage: BootStrapClusters
-    displayName: 'BootStrap Clusters'
-    dependsOn: Aks
-    jobs:
-      - job: BootStrap
+      - stage: "Aks_${{ component.env }}"
+        displayName: "${{ upper(component.env) }}: AKS clusters"
+        dependsOn:
+          - "CheckingClusters_${{ component.env }}"
+          - "Managed_Identity_${{ component.env }}"
+        variables:
+          existing_clusters: $[stageDependencies.CheckingClusters_${{ component.env }}.CheckingClusters.outputs['check_clusters.cluster_deploy']]
+          deploy_cluster: ${{ parameters.cluster }}
         condition: |
           or(
-            and(succeeded(), eq('${{ parameters.action }}', 'apply')),
-            and(succeeded(), eq(variables['isMain'], true), eq(variables['isAutoTriggered'], true))
+            and(eq('${{ parameters.cluster }}', 'All'), eq(variables['Build.Reason'], 'Manual')),
+            and(in(dependencies.CheckingClusters_${{ component.env }}.outputs['CheckingClusters.check_clusters.cluster_deploy'], 'All', '00', '01'), eq(variables['Build.Reason'], 'Manual')),
+            and(eq(stageDependencies.CheckingClusters_${{ component.env }}.CheckingClusters.outputs['check_clusters.cluster_deploy'], 'All'), eq(variables['Build.Reason'], 'Manual'))
           )
-        variables:
-          clusters: ${{ parameters.cluster }}
-        steps:
-          - template: pipeline-steps/bootstrap.yaml
-            parameters:
-              environment: ${{ parameters.env }}
-              cluster: ${{ parameters.cluster }}
-              project: $(project)
-              serviceConnection: $(serviceConnection)
-              aksKeyVault: $(aksKeyVault)
-              terraformSubscriptionID : $(terraformSubscriptionID)
+        jobs:
+          - job: SetClusterVariables
+            steps:
+              - script: echo "##vso[task.setvariable variable=existing_clusters;isOutput=true]value"
+                name: SetExistingClustersVariable
+              - script: echo "##vso[task.setvariable variable=deploy_cluster;isOutput=true]value"
+                name: SetDeployClusterVariable
+          - job: DeployInfrastructure
+            condition: |
+              or(
+                and(succeeded(), eq(variables['isAutoTriggered'], true), eq(variables['Build.Reason'], 'PullRequest')),
+                and(succeeded(), eq(variables['isMain'], true), eq(variables['isAutoTriggered'], true))
+              )
```

Comment:
What are these conditions for? Doesn't make much sense to me the need for them here - shouldn't the condition be in relation to those cluster variables you have?

Reply:
/ stuff like this seems to belong to the stage too

Reply:
removed now